src/isa.rs

```diff
@@ -332,35 +537,35 @@ impl Instructions {
 		self.vec.len() as u32
 	}
 
-	pub fn push(&mut self, instruction: Instruction) {
+	pub(crate) fn push(&mut self, instruction: InstructionInternal) {
```

Comment:
TODO: We might want to make it so you supply a `Instruction` and we convert it to an `InstructionInternal`, then have a `pub(crate) fn push_internal`